bdk_core/src/byte_list.rs

```diff
@@ -2,13 +2,13 @@ use alloc::vec::Vec;
 use core::mem::size_of;
 use std::io::{self, Read, Seek, SeekFrom, Write};
 
-/// A very simple collection of `256` on disk linked lists.
+/// A very simple collection of `256` byte linked lists.
 #[derive(Debug)]
-pub struct DiskList<F> {
-    inner: F,
+pub struct ByteList<S> {
```

Comment:
It's not a list of bytes :). I suppose it's not a list of disks either.

Just call it `ExampleDb` or something. It probably shouldn't leave this repo.